javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentInstaller.java

```diff
@@ -122,8 +125,14 @@ private static void installBytebuddyAgent(
     }
 
     AgentBuilder agentBuilder =
-        new AgentBuilder.Default()
+        new AgentBuilder.Default(
+                // default method graph compiler inspects the class hierarchy, we don't need it, so
+                // we use a simpler and faster strategy instead
+                new ByteBuddy().with(MethodGraph.Compiler.ForDeclaredMethods.INSTANCE))
```

Comment:
Method graph compiler is used in https://github.com/raphw/byte-buddy/blob/0308dfed5be3beebe9ca777fc891b78b6ecf058b/byte-buddy-dep/src/main/java/net/bytebuddy/dynamic/scaffold/MethodRegistry.java#L450 I don't pretend to fully understand why this is needed, but I suspect that as we are not doing structural modifications to the class we don't really need it and can thus use the faster variant of method graph compiler.